src/main/java/engineering/swat/watch/Watcher.java

```diff
@@ -99,13 +99,44 @@ public static Watcher watch(Path path, WatchScope scope) {
      * @param eventHandler a callback that handles the watch event, will be called once per event.
      * @return this for optional method chaining
      */
-    public Watcher onEvent(Consumer<WatchEvent> eventHandler) {
+    public Watcher on(Consumer<WatchEvent> eventHandler) {
+        if (this.eventHandler != NULL_HANDLER) {
```

Comment:
Nit: Not really related to this PR, but `EMPTY_HANDLER` is maybe a more accurate name than `NULL_HANDLER`. (`NULL_HANDLER` could be misread as "a handler dedicated to handling `null` events").

Reply:
Agreed, fixed.